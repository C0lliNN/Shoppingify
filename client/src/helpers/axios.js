import axios from 'axios';
import store from '../store/index';

export default function getAxios() {
  const config = {
    headers: {},
  };
  // eslint-disable-next-line no-undef
  if (process.env.NODE_ENV === 'development') {
    config.baseURL = 'http://localhost:5000/api/v1';
  } else {
    config.baseURL = '/api/v1';
  }

  config.headers['Content-Type'] = 'application/json';

  const token = store.getState().auth.token;

  if (token) {
    config.headers['Authorization'] = `Bearer ${token}`;
  }

  return axios.create(config);
}
