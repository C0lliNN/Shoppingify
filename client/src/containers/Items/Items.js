import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import Title from '../../components/Title';
import SearchBar from '../../components/UI/SearchBar/SearchBar';
import { useEffect } from 'react';
import ItemsGroup from '../../components/ItemsGroup';
import * as variables from '../../helpers/style-constants';
import { getItemsData } from '../../store/actions';
import { connect } from 'react-redux';
import Spinner from '../../components/UI/Spinner/Spinner';
import ErrorMessage from '../../components/ErrorMessage';

const StyledItems = styled.div`
  padding: 20px;
  box-sizing: border-box;
  min-height: 100%;
  background-color: #fafafe;
  @media (min-width: ${variables.MD_BREAK_POINT}px) {
    padding: 20px 60px;
  }
`;

const Header = styled.div`
  display: flex;
  align-items: top;
  justify-content: start;
  flex-wrap: wrap;
  @media (min-width: ${variables.MD_BREAK_POINT}px) {
    flex-wrap: nowrap;
    justify-content: space-around;
  }
`;

const Text = styled.p`
  font-size: ${variables.FONT_SIZE_3};
  font-family: ${variables.FONT_FAMILY};
  font-weight: 600;
  margin-top: 50px;
`;

function Items({ getItemsData, data, isLoading, error }) {
  useEffect(() => {
    getItemsData();
  }, [getItemsData]);

  let content = null;

  if (isLoading) {
    content = <Spinner />;
  } else if (error) {
    content = <ErrorMessage message={error} />;
  } else if (data.length) {
    content = data
      .filter((group) => group.category._id)
      .map((group) => <ItemsGroup key={group.category._id} {...group} />);
  } else {
    content = <Text>No items yet</Text>;
  }

  return (
    <StyledItems>
      <Header>
        <Title />
        <SearchBar />
      </Header>
      {content}
    </StyledItems>
  );
}

Items.propTypes = {
  isLoading: PropTypes.bool,
  error: PropTypes.string,
  data: PropTypes.array,
  getItemsData: PropTypes.func,
};

const mapDispatchToProps = {
  getItemsData,
};

const mapStateToProps = (state) => {
  return {
    ...state.itemsData,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Items);
