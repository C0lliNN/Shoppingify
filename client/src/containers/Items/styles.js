import styled from 'styled-components';
import * as variables from '../../helpers/style-constants';

export const StyledItems = styled.div`
  padding: 20px;
  box-sizing: border-box;
  min-height: 100%;
  background-color: #fafafe;
  @media (min-width: ${variables.MD_BREAK_POINT}px) {
    padding: 20px 60px;
  }
`;

export const Header = styled.div`
  display: flex;
  align-items: top;
  justify-content: start;
  flex-wrap: wrap;
  @media (min-width: ${variables.MD_BREAK_POINT}px) {
    flex-wrap: nowrap;
    justify-content: space-around;
  }
`;

export const Text = styled.p`
  font-size: ${variables.FONT_SIZE_3};
  font-family: ${variables.FONT_FAMILY};
  font-weight: 600;
  margin-top: 50px;
`;