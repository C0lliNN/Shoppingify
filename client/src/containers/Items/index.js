import React from 'react';
import Title from '../../components/Title';
import SearchBar from '../../components/UI/SearchBar';
import ItemsGroup from '../../components/ItemsGroup';
import { useSelector } from 'react-redux';
import Spinner from '../../components/UI/Spinner';
import ErrorMessage from '../../components/ErrorMessage';
import { Text, StyledItems, Header } from './styles';

function itemsReducer(items, group) {
  return items.concat(group.items.map((item) => item._id));
}

function Items() {
  const { data, isLoading, error } = useSelector((state) => state.itemsData);
  const itemsAdded = useSelector((state) =>
    state.activeList.data.reduce(itemsReducer, [])
  );

  let content = null;

  if (isLoading) {
    content = <Spinner />;
  } else if (error) {
    content = <ErrorMessage message={error} />;
  } else if (data.length) {
    content = data
      .filter((group) => group.category._id)
      .map((group) => (
        <ItemsGroup
          key={group.category._id}
          category={group.category}
          items={group.items.filter((item) => !itemsAdded.includes(item._id))}
        />
      ));
  } else {
    content = <Text>No items yet</Text>;
  }

  return (
    <StyledItems>
      <Header>
        <Title />
        <SearchBar />
      </Header>
      {content}
    </StyledItems>
  );
}

export default Items;
