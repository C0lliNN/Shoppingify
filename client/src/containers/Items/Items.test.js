import React from 'react';
import { render, getByTextWithMarkup } from '../../tests/utilities';
import Items from './Items';
import axios from 'axios';
import { waitForElement } from '@testing-library/react';
import 'jest-styled-components';

function exec() {
  return render(<Items />);
}

jest.mock('axios');

describe('<Items/>', () => {
  beforeEach(() => {
    axios.create.mockImplementation(() => ({
      get: jest.fn().mockResolvedValue({
        data: [
          {
            _id: '5f5543b2267b3e04c0a91680',
            name: 'Apple',
            note: '',
            image: '',
            category: { _id: '5f5543ab267b3e04c0a9167f', name: 'Fruit' },
            user: '5f528e6790213f0b78431156',
            __v: 0,
          },
          {
            _id: '5f5543ab267b3e04c0a9167e',
            name: 'Banana',
            note: '',
            image: '',
            category: { _id: '5f5543ab267b3e04c0a9167f', name: 'Fruit' },
            user: '5f528e6790213f0b78431156',
            __v: 0,
          },
        ],
      }),
    }));
  });
  it('should render correctly', async () => {
    const { container, getByText } = exec();

    await waitForElement(() => getByText('Apple'));
    expect(container).toMatchSnapshot();
  });
  it('should display the Title the title', () => {
    const { getByText } = exec();

    const element = getByTextWithMarkup(
      getByText,
      'Shoppingify allows you take your shopping list wherever you go'
    );

    expect(element).not.toBeNull();
  });
  it('should display the category', async () => {
    const { getByText } = exec();

    await waitForElement(() => getByText('Fruit'));
  });
  it('should display all items', async () => {
    const { getByText } = exec();

    await waitForElement(() => getByText('Apple'));
    expect(getByText('Banana')).not.toBeNull();
  });
});
