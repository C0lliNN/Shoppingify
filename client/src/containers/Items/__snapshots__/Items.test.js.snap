// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Items/> should render correctly 1`] = `
<div>
  <div
    class="sc-fznKkj fNqfXh"
  >
    <div
      class="sc-fznZeY NaBzt"
    >
      <h1
        class="sc-AxjAm GKfhy"
      >
        <span
          style="color: rgb(249, 161, 9); font-weight: bold;"
        >
          Shoppingify
           
        </span>
        allows you take your shopping list wherever you go
      </h1>
      <div
        class="sc-AxhUy hdwioC"
      >
        <div
          class="sc-AxirZ fuDjKo"
        >
          <div
            class="sc-AxiKw fqTuty material-icons-round"
          >
            search
          </div>
        </div>
        <input
          class="sc-AxhCb UzUCm"
          placeholder="Search item"
        />
      </div>
    </div>
    <div
      style="margin-top: 50px;"
    >
      <h3
        class="sc-AxgMl ccGqat"
      >
        Fruit
      </h3>
      <div
        class="sc-AxheI cAytUM"
      >
        <div
          class="sc-Axmtr hzHwRp"
        >
          <span
            class="sc-AxmLO ceQsip"
          >
            Apple
          </span>
          <i
            class="sc-fzozJi jaJZqY material-icons-round"
          >
            add
          </i>
        </div>
        <div
          class="sc-Axmtr hzHwRp"
        >
          <span
            class="sc-AxmLO ceQsip"
          >
            Banana
          </span>
          <i
            class="sc-fzozJi jaJZqY material-icons-round"
          >
            add
          </i>
        </div>
      </div>
    </div>
  </div>
</div>
`;
